.travis.yml

```diff
@@ -1,8 +1,50 @@
+sudo: false
+
 language: php
-php:  
-  - 5.6
-  - 7.0
 
-before_script: composer install
+cache:
+  directories:
+    - $HOME/.composer/cache
+
+env:
+  global:
+    - COMPOSER_ARGS="--no-interaction"
+
+matrix:
+  include:
+    - php: 5.6
+      env:
+        - DEPS=lowest
+    - php: 5.6
+      env:
+        - DEPS=latest
+    - php: 7
+      env:
+        - DEPS=lowest
+    - php: 7
+      env:
+        - DEPS=latest
+    - php: 7.1
+      env:
+        - DEPS=lowest
+    - php: 7.1
+      env:
+        - DEPS=latest
+    - php: 7.2
+      env:
+        - DEPS=lowest
+    - php: 7.2
+      env:
+        - DEPS=latest
+  allow_failures:
+    - php: 7.2
+
+before_install:
+  - travis_retry composer self-update
+
+install:
+  - if [[ $DEPS == 'latest' ]]; then travis_retry composer update $COMPOSER_ARGS ; fi
+  - if [[ $DEPS == 'lowest' ]]; then travis_retry composer update --prefer-lowest --prefer-stable $COMPOSER_ARGS ; fi
```

Comment:
I wasn't aware of this prefer-lowest . Could you tell this lazy guy why it is used ?

Reply:
hehe, composer will resolve the lowest possible dependencies. Suppose we require ^2.2.4 of some package and 2.3.6 has been released. It will use 2.3.6 with as normal and 2.2.4 with the --lowest flag